provisioning/app-proxy/kubernetes/templates/proxy/deployment.yaml

```diff
@@ -0,0 +1,117 @@
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: app-proxy
+  namespace: $NAMESPACE
+  labels:
+    app: app-proxy
+    releaseId: $RELEASE_ID
+spec:
+  replicas: $APP_PROXY_REPLICAS
+  selector:
+    matchLabels:
+      app: app-proxy
+  template:
+    metadata:
+      labels:
+        app: app-proxy
+        releaseId: $RELEASE_ID
+        tags.datadoghq.com/env: "$KEBOOLA_STACK"
+        tags.datadoghq.com/service: "app-proxy"
+        tags.datadoghq.com/version: "$RELEASE_ID"
+      annotations:
+        log: "true"
+        ad.datadoghq.com/tags: '{"git.commit.sha": "$COMMIT_HASH", "git.repository_url": "github.com/keboola/keboola-as-code"}'
+        ad.datadoghq.com/app-proxy.checks: |
+          {
+            "openmetrics": {
+              "instances": [
+                {
+                  "openmetrics_endpoint": "http://%%host%%:9000/metrics",
+                  "metrics": [".*"],
+                  "collect_histogram_buckets": true,
+                  "histogram_buckets_as_distributions": true,
+                  "tag_by_endpoint": false
+                }
+              ]
+            }
+          }
+    spec:
+      containers:
+        - name: app-proxy
+          image: $APP_PROXY_REPOSITORY:$APP_PROXY_IMAGE_TAG
+          resources:
+            requests:
+              cpu: "200m"
+              memory: "768Mi"
```

Comment:
memory request (tj pamat s ktorou pocita k8s scheduler pri alokacii podu na node) by som znizil, [vidim](https://app.datadoghq.eu/dashboard/gny-t8t-d99/kubernetes?fromUser=true&fullscreen_end_ts=1709031416093&fullscreen_paused=false&fullscreen_refresh_mode=sliding&fullscreen_section=overview&fullscreen_start_ts=1708945016093&fullscreen_widget=6879185011415216&refresh_mode=sliding&tpl_var_kube_namespace%5B0%5D=app-proxy&view=spans&from_ts=1708945014095&to_ts=1709031414095&live=true) ze to realne zabera okolo 30MB, tak request na 50MB namiesto 768MB mi pride vhodnejsi, navyse limit je 768MB.

Reply:
ak sa naopak pocita s tym ze produkcny load vyzaduje prave tolko vela pamate, tj konstantne okolo 768MB, tak by som dvihol limit na 1GB.

Reply:
Este sme to neskusali pod zatazou, pod `500MB` (limit) by som nesiel. Staci nam to v 2 instanciach.
Vacsinou ste z SRE chceli, aby bol request a limit rovnaky. Ak sa to zmenilo, tak request samozrejme moze ist dolu.

Reply:
> Vacsinou ste z SRE chceli, aby bol request a limit rovnaky.

aha, to si nespominam, neviem si teraz predstavit kde by to mohlo davat zmysel okrem pripadu ked to je konstantne od zaciatku velmi vysoke. Request je to s cim pocita scheduler ked sklada pody na nody, limit je hranica pri ktorej terminuje pody. V [service requirements](https://keboola.atlassian.net/wiki/spaces/ENGG/pages/1453785161/Service+requirements#Resource-limits-%26-requests) je k tomu napisane ze request ma byt max 75% limit. Ak vam pride rozumny min limit 500MB tak request by som dal 150MB, alebo teda pri 768MB limite 200MB request. 
Kazdopadne mozeme to sledovat a postupne upravovat.

Reply:
:heavy_check_mark: 

ok, dava zmysel, je mi jasne ako to funguje, len mi nebolo jasne ake hodnoty tam chcete

> aha, to si nespominam, 

nejdem asi hladat PR starsie ako 1 rok, ale z toho dovodu sme davali vsade rovnake hodnoty

> V service requirements je k tomu napisane ze request ma byt max 75% limit. 

**[DOCS](https://keboola.atlassian.net/wiki/spaces/ENGG/pages/1453785161/Service+requirements#Resource-limits-%26-requests): Requesty by nemělý být míň než ~75% limitů, jinak se může stát, že instance bude busy a začne ukončovat kontejnery.** 

@kacurez toto nesedi ^^^ `nemělý být míň než ~75% limitů`, `500*0.75=375MB`, ty pises `150MB` - tj. to sedi s tou mojou predstavou, ze sme to davali rovnake (resp nie pod 75%).

Teda ako to ma byt? :slightly_smiling_face:

Reply:
j ja som to bral opacne a zle som si to vypocital, [tu je napr prehlad limitov a requestov api deploymentov](https://app.datadoghq.eu/dashboard/gny-t8t-d99/kubernetes?fromUser=false&fullscreen_end_ts=1709033792675&fullscreen_paused=false&fullscreen_refresh_mode=sliding&fullscreen_section=overview&fullscreen_start_ts=1708947392675&fullscreen_widget=6678268604183702&refresh_mode=sliding&tpl_var_env%5B0%5D=kbc-testing-azure-east-us-2&tpl_var_kube_deployment%5B0%5D=%2Aapi&tpl_var_kube_namespace%5B0%5D=%2A&view=spans&from_ts=1708947302353&to_ts=1709033702353&live=true) - vacsina z nich ma ten pomer okolo 50% aj menej, tak teraz neviem odkial sa to cislo >75% berie :D 
![image](https://github.com/keboola/keboola-as-code/assets/1412120/18211ab6-c014-475e-8053-39d859839838)

Tiez napriklad [buffer-api](https://app.datadoghq.eu/dashboard/gny-t8t-d99/kubernetes?fromUser=false&fullscreen_end_ts=1709033866628&fullscreen_paused=false&fullscreen_refresh_mode=sliding&fullscreen_section=overview&fullscreen_start_ts=1706441866628&fullscreen_widget=6879185011415216&refresh_mode=sliding&tpl_var_kube_deployment%5B0%5D=buffer-api&tpl_var_kube_namespace%5B0%5D=%2A&view=spans&from_ts=1708947302353&to_ts=1709033702353&live=true) (req = limit = 512MB) za posledny mesiac malo peak v US 470MB a EU 250MB ale v ostatnych stackoch je vacsinou pod 200MB. 
Ale ok, ak nevieme kolko to bude zrat pamate tak mozme to na zaciatok nechat vysoko.

Reply:
@kacurez a zhodneme sa na tom, ze si to vypocital inak ako je to v docs? :slightly_smiling_face: 

Mozeme tam dat kludne `150MB`, len to prosim fixnime/upravne aj v docs, ... nech nepride o mesiac niekto iny z SRE teamu a nepyta sa, "preco to nemate podla nasich standardov?".

Reply:
jj, tak dajte to prosim na 50% z limitu. To aky limit, necham na vas.
S tymi standardami som si nabehol no :) Poznacim si to bokom a preberiem s SRE.

Reply:
prosel jsem si, doc je cajk. ne vsechno ji ale respektuje a ne vsude to dava smysl. nechte to takhle zatim.